src/track/taglib/trackmetadata_ape.cpp

```diff
@@ -146,6 +156,9 @@ void importTrackMetadataFromTag(TrackMetadata* pTrackMetadata, const TagLib::APE
                 &discTotal);
         pTrackMetadata->refTrackInfo().setDiscNumber(discNumber);
         pTrackMetadata->refTrackInfo().setDiscTotal(discTotal);
+    } else if (resetMissingTagMetadata) {
+        pTrackMetadata->refTrackInfo().setDiscNumber(QString{});
+        pTrackMetadata->refTrackInfo().setDiscTotal(QString{});
     }
 #endif // __EXTRA_METADATA__
 
```

Comment:
Just curious, below we also parse bpm, trackgain, etc. Why are we not resetting those?

Reply:
Would have been extra work and for those the tags don't matter. You could reset them in Mixxx. Slightly inconsistent but acceptable.

Reply:
Sure. Would you mind adding a that sentence as a comment above the imports you didn't touch?

Reply:
A comment was already in place at the entry point. I have extended it: https://github.com/mixxxdj/mixxx/pull/4873/commits/5d1f73b13411de04f7cd89fbc8b09a227455da65

Reply:
Sorry I seem to have read past that. Thanks.

Reply:
Unfortunately, the code is a little bit scattered and the entry point is not obvious.

The boolean could be transformed into an enum for type safety, but the boolean was easier to handle while extending the implementation. That was a deliberate decision.

Reply:
I have decided to add the missing reset of both bpm and replay gain: https://github.com/mixxxdj/mixxx/pull/4873/commits/c8487255c58f3f4355e673e5404b059d1cb9ac14